sdk/python/kfp/compiler/compiler.py

```diff
@@ -182,7 +184,86 @@ def _get_condition_params_for_ops_helper(group, current_conditions_params):
     _get_condition_params_for_ops_helper(root_group, [])
     return conditions
 
-  def _get_inputs_outputs(self, pipeline, root_group, op_groups, opsgroup_groups, condition_params):
+  def _get_next_group_or_op(cls, to_visit: List, already_visited: Set):
+    """Get next group or op to visit."""
+    if len(to_visit) == 0:
+      return None
+    next = to_visit.pop(0)
+    while next in already_visited:
+      next = to_visit.pop(0)
+    already_visited.add(next)
+    return next
+
+  def _get_for_loop_ops(self, new_root) -> Dict[Text, dsl.ParallelFor]:
+    to_visit = self._get_subgroups(new_root)
+    op_name_to_op = {}
+    already_visited = set()
+
+    while len(to_visit):
+      next_op = self._get_next_group_or_op(to_visit, already_visited)
+      if next_op is None:
+        break
+      to_visit.extend(self._get_subgroups(next_op))
+      if isinstance(next_op, dsl.ParallelFor):
+        op_name_to_op[next_op.name] = next_op
+
+    return op_name_to_op
+
+  def _fill_loop_args(self, new_root):
+    """Traverses through graph, plucking up loop_args vars from ops groups and depositing pointers to them on the
+    ops which contain them as arguments."""
+
+    def get_loop_args_names_for_op(op: _container_op.BaseOp):
+      if not isinstance(op, _container_op.BaseOp):
+        return []
+
+      loop_arg_names = []
+      for pipeline_param in op.inputs:
+        # make sure to keep this in this order since both LoopArgumentVariable.param_is_this_type and
+        # LoopArgument.param_is_this_type will return true for LoopArgumentVariable parameters
+        if dsl.LoopArgumentVariable.param_is_this_type(pipeline_param):
+          loop_args_name, _ = dsl.LoopArgumentVariable.parse_loop_args_name_and_this_var_name(pipeline_param.name)
+          loop_args_names.append(loop_args_name)
+        elif dsl.LoopArguments.param_is_this_type(pipeline_param):
+          loop_arg_names.append(pipeline_param.name)
+
+      return loop_arg_names
+
+    to_visit = self._get_subgroups(new_root)
+    loop_args_name_to_obj = {}
+    already_visited = set()
+
+    while len(to_visit):
+      next = self._get_next_group_or_op(to_visit, already_visited)
+      if next is None:
+        break
+      to_visit.extend(self._get_subgroups(next))
+
+      # store loop_args from ops groups
+      if hasattr(next, 'loop_args') and next.loop_args is not None:
+        loop_args_name_to_obj[next.loop_args.name] = next.loop_args
+
+      loop_args_names = get_loop_args_names_for_op(next)
+      next.loop_args_names_to_loop_args = {name: loop_args_name_to_obj[name] for name in loop_args_names}
```

Comment:
Where is this used?

Reply:
it isn't.  in fact this whole function is out of date; I'll remove it.